coreneuron/gpu/nrn_acc_manager.cpp

```diff
@@ -787,6 +814,19 @@ void update_voltage_from_gpu(NrnThread* nt) {
     }
 }
 
+/**
+ * @brief Copy fast_imem vectors from GPU to CPU.
+ *
+ */
+void update_fast_imem_from_gpu(NrnThread* nt) {
+    if (nt->compute_gpu && nt->end > 0 && nt->nrn_fast_imem) {
+        int num_fast_imem = nt->end;
+        double* fast_imem_d = nt->nrn_fast_imem->nrn_sav_d;
+        double* fast_imem_rhs = nt->nrn_fast_imem->nrn_sav_rhs;
+#pragma acc update host(fast_imem_d [0:num_fast_imem], fast_imem_rhs [0:num_fast_imem])
```

Comment:
Just as a note: in the old versions of PGI/Cray compiler we had issue of pointer dereference in `pragma`.

Reply:
OK, maybe we can remove some of these temporary variables if/when we change OpenACC -> OpenMP.